### PR TITLE
Improve mypy support

### PR DESCRIPTION
**Pull request by morgangiraud**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -66,6 +67,11 @@ test: lint pythoncheck
 lint: pythoncheck
 	flake8
 
+typecheck: pythoncheck
```
Comment (ianlivingstone) [inline Makefile]: We should add this to the `make test` command right?

It should probably also be specified in the `.PHONY`

Comment (morgangiraud) [reply Makefile]: Yes for `.PHONY`

Not yet for make test, because it breaks too much for now

Comment (morgangiraud) [reply Makefile]: I actually think, someone should focus on mypy for 2 days, fix everything ad add it to `make test` and force other contributors to comply afterward.

Comment (ianlivingstone) [reply Makefile]: How about we land this in master and then open another issue for converting the rest of the code base?

Comment (morgangiraud) [reply Makefile]: Agree

Comment (ianlivingstone) [reply Makefile]: 👍 Let's get the PHONY updated and an issue, then I'll 🍏 :)

**Pull request merged**